atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1460,14 +1446,7 @@ protected void throwIfReadWriteConflictForSerializable(long commitTimestamp) {
     }
 
     private boolean hasWrites() {
-        boolean hasWrites = false;
-        for (SortedMap<?, ?> map : writesByTable.values()) {
-            if (!map.isEmpty()) {
-                hasWrites = true;
-                break;
-            }
-        }
-        return hasWrites;
+        return writesByTable.values().stream().anyMatch(writesForTable -> !writesForTable.isEmpty());
```

Comment:
are we opposed to doing `hasWrites` as a `volatile boolean` similar to `hasReads`?

Reply:
Not opposed if you think that's clearer, although this is also sufficient. There's already a lot of state being tracked in this class.

Reply:
I shall leave it, agree with the state tracking bit.